src/backends/torch/torchmodule.cc

```diff
@@ -222,6 +222,26 @@ namespace dd
       const torch::Device &device, const APIData &ad)
   {
     post_transform(tmpl, template_params, inputc, tmodel, device);
+
+    if (_require_linear_layer && !_linear)
+      {
+        try
+          {
+            // TODO const cast because getting an input example actually
+            // *modifies* the connector (it must be reset after that)
+            // -> find a way to get an example without modifying the dataset?
+            setup_linear_layer(_nclasses,
+                               const_cast<TInputConnectorStrategy &>(inputc)
+                                   .get_input_example(device));
+            _linear->to(_device);
+          }
+        catch (std::exception &e)
+          {
+            throw MLLibInternalException(std::string("Libtorch error: ")
+                                         + e.what());
+          }
+      }
```

Comment:
If this is both in `post_transform_train` and `post_transform_predict` it can go in `post_transform` instead